pkg/arvo/sys/zuse.hoon

```diff
@@ -6084,4 +6084,1498 @@
   ?.  ?=(%soft -.wrapped)
     wrapped
   ;;(task +.wrapped)
+::                                                    ::
+::::                    ++la                          ::  (2v) vector/matrix ops
+++  la
+  |%
+    ::                                                  ::::
+  ::  Single-precision floating-point vector type & operations
+  ::
+  ++  lvs
+    ^|
+    |_  r=$?(%n %u %d %z)   :: round nearest, round up, round down, round to zero
+    ::
+    ::  Manipulators
+    ::
+    ::    Zeroes
+    ++  zeros
+      |=  n=@ud  ^-  @lvs
+      ~_  leaf+"lagoon-fail"
+      `@lvs`(lsh [5 n] 1)   :: pin at head for leading zeros
+    ::
+    ::    Fill value
+    ++  fill
+      |=  [n=@ud s=@rs]  ^-  @lvs
+      `@lvs`(mix (zeros n) (fil 5 n s))
+    ::
+    ::    Ones
+    ++  ones
+      |=  n=@ud  ^-  @lvs
+      ~_  leaf+"lagoon-fail"
+      (fill n .1)
+    ::
+    ::    Length of vector
+    ++  length
+      |=  u=@lvs  ^-  @ud
+      ~_  leaf+"lagoon-fail"
+      =/  ell  (met 5 u)
+      ?:  (gth ell 1)  (dec ell)  0
```

Comment:
Can't this always be `(dec ell)`?

Reply:
`` `@lvd`0x0`` is permitted.

Reply:
We can revisit that when we switch to the cell format.